### PR TITLE
Make Pod mutator idempotent to support fluid

### LINKED ISSUE #2895: KServe not compatible with fluid
/kind bug

**What steps did you take and what happened:**
KServe is not compatible with fluid. When using fluid to load model for InferenceService, kserve controller will throw an error as below:
```
kserve-fluid-demo   2m33s       Warning   InternalError           pod/fluid-bloom-predictor-00001-deployment-5b6cf84f44-7whjh   Pod "fluid-bloom-predictor-00001-deployment-5b6cf84f44-7whjh" is invalid: [spec.volumes[5].name: Duplicate value: "kserve-pvc-source", spec.containers[1].volumeMounts[3].mountPath: Invalid value: "/mnt/models": must be unique, spec: Forbidden: pod updates may not change fields other than `spec.containers[*].image`, `spec.initContainers[*].image`, `spec.activeDeadlineSeconds`, `spec.tolerations` (only additions to existing tolerations) or `spec.terminationGracePeriodSeconds` (allow it to be set to 1 if it was previously negative)
  core.PodSpec{
      Volumes: []core.Volume{
        ... // 3 identical elements
        {Name: "bigboot-config-0", VolumeSource: {ConfigMap: &{LocalObjectReference: {Name: "oss-data-jindofs-config"}, DefaultMode: &420}}},
        {Name: "check-mount-unprivileged-0", VolumeSource: {ConfigMap: &{LocalObjectReference: {Name: "oss-data-jindo-check-mount-unprivileged"}, DefaultMode: &493}}},
+          {
+             Name: "kserve-pvc-source",
+             VolumeSource: core.VolumeSource{
+             PersistentVolumeClaim: &core.PersistentVolumeClaimVolumeSource{ClaimName: "oss-data"},
+             },
+          },
      },
      InitContainers: nil,
      Containers: []core.Container{
        {Name: "fluid-fuse-0", Image: "registry-vpc.cn-hangzhou.aliyuncs.com/jindofs/jindo-fuse:4.5.2", Command: {"/entrypoint.sh"}, Args: {"-okernel_cache", "-oro", "-oattr_timeout=7200", "-oentry_timeout=7200", ...}, ...},
        {
          ... // 7 identical fields
          Env:       {{Name: "STORAGE_URI", Value: "/mnt/models"}, {Name: "MODEL_NAME", Value: "bloom"}, {Name: "GPU_ENABLED", Value: "True"}, {Name: "PORT", Value: "8080"}, ...},
          Resources: {Limits: {s"cpu": {i: {...}, s: "12", Format: "DecimalSI"}, s"memory": {i: {...}, Format: "BinarySI"}}, Requests: {s"cpu": {i: {...}, s: "12", Format: "DecimalSI"}, s"memory": {i: {...}, Format: "BinarySI"}}},
          VolumeMounts: []core.VolumeMount{
            {Name: "kube-api-access-n6cdk", ReadOnly: true, MountPath: "/var/run/secrets/kubernetes.io/serviceaccount"},
            {Name: "kserve-pvc-source", ReadOnly: true, MountPath: "/mnt/models", SubPath: "bloom-560m", ...},
            {Name: "check-fluid-mount-ready", ReadOnly: true, MountPath: "/check-fluid-mount-ready.sh", SubPath: "check-fluid-mount-ready.sh", ...},
+             {
+               Name:      "kserve-pvc-source",
+               ReadOnly:  true,
+               MountPath: "/mnt/models",
+               SubPath:   "bloom-560m",
+             },
          },
          VolumeDevices: nil,
          LivenessProbe: nil,
          ... // 10 identical fields
        },
        {Name: "queue-proxy", Image: "registry-vpc.cn-hangzhou.aliyuncs.com/acs/serving-cmd-queue:2022"..., Ports: {{Name: "http-queueadm", ContainerPort: 8022, Protocol: "TCP"}, {Name: "http-autometric", ContainerPort: 9090, Protocol: "TCP"}, {Name: "http-usermetric", ContainerPort: 9091, Protocol: "TCP"}, {Name: "queue-port", ContainerPort: 8012, Protocol: "TCP"}, ...}, Env: {{Name: "SERVING_NAMESPACE", Value: "kserve-fluid-demo"}, {Name: "SERVING_SERVICE", Value: "fluid-bloom-predictor"}, {Name: "SERVING_CONFIGURATION", Value: "fluid-bloom-predictor"}, {Name: "SERVING_REVISION", Value: "fluid-bloom-predictor-00001"}, ...}, ...},
      },
      EphemeralContainers: nil,
      RestartPolicy:       "Always",
      ... // 26 identical fields
  }
]
```

**I think the  possible reason is:**
First, the webhook of kserve adds the PVC declared in STORAGE_URI as a volume of the container, and then it triggers the webhook of fluid to convert the PVC to hostPath. This behavior then triggers the watch for Pod UPDATE events in kserve webhook, which detects that the PVC is not mounted and tries to add a volume with the same name, resulting in an error.

The corresponding solution is to remove the watch for Pod UPDATE event in MutatingWebhookConfiguration `inferenceservice.serving.kserve.io` 
https://github.com/kserve/kserve/blob/5900a68c398526996294e5647d436fd4d9d4877c/charts/kserve-resources/templates/webhookconfiguration.yaml#L43-L52

**What did you expect to happen:**
KServe is compatible with fluid.

**What's the InferenceService yaml:**
[To help us debug please run `kubectl get isvc $name -n $namespace -oyaml` and paste the output]

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Istio Version: v1.17
- Knative Version: v1.4.2
- KServe Version: build from master, commit ID: a224b740f31c7c9d9032ba257483f6fa869afb61
- Kubeflow version:
- Cloud Environment:[k8s_istio/istio_dex/gcp_basic_auth/gcp_iap/aws/aws_cognito/ibm]
- Minikube/Kind version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by hclchimumu**
**What this PR does / why we need it**:
Remove the watch for Pod UPDATE event in webhook `inferenceservice.serving.kserve.io` to make KServe compatible with fluid. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2895

**Type of changes**
Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevent result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [x] follow this doc https://github.com/kserve/kserve/blob/master/docs/samples/fluid/README.md
- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.-No

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note
Remove the watch for Pod UPDATE event in webhook to support fluid
```

### CODE REVIEW

**Comment by yuzisun** [comment]
@hclchimumu can you help sign off the commit ?

**Comment by hclchimumu** [comment]
> @hclchimumu can you help sign off the commit ?

signed

**Comment by yuzisun** [comment]
@hclchimumu kserve pod mutator also may not be first pod mutating webhook, so I think we still need to watch the update event, instead we should make the kserve pod mutator idempotent. Before appending the volume we can check if the volume exists already here, cc @lizzzcai 
https://github.com/kserve/kserve/blob/feea1354a09322eda477c8ee615b324df969e83e/pkg/webhook/admission/pod/storage_initializer_injector.go#L157

**Comment by hclchimumu** [comment]
I have update my code. When using direct volume mount, add check before append VolumeMounts/Volume to make it idempotent. @yuzisun @lizzzcai

File: pkg/webhook/admission/pod/storage_initializer_injector.go [hunk 1]
```diff
@@ -154,6 +154,17 @@ func (mi *StorageInitializerInjector) InjectStorageInitializer(pod *v1.Pod) erro
...
+				for _, volumeMount := range userContainer.VolumeMounts {
+					if strings.Compare(volumeMount.Name, PvcSourceMountName) == 0 {
+						return nil
+					}
+				}
+			}
+
 			userContainer.VolumeMounts = append(userContainer.VolumeMounts, pvcSourceVolumeMount)
 			if transformerContainer != nil {
 				transformerContainer.VolumeMounts = append(transformerContainer.VolumeMounts, pvcSourceVolumeMount)
```
Comment (lizzzcai) [inline pkg/webhook/admission/pod/storage_initializer_injector.go:179]: I think the transformer container probably need to check as well.

Comment (hclchimumu) [reply pkg/webhook/admission/pod/storage_initializer_injector.go:179]: Sure. I have updated the code. @lizzzcai

**Comment by yuzisun** [comment]
Thanks @hclchimumu !

/lgtm
/approve

**Pull request merged**